src/sentry/static/sentry/app/views/organizationIncidents/details/activity/index.jsx

```diff
@@ -0,0 +1,195 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import {INCIDENT_ACTIVITY_TYPE} from 'app/views/organizationIncidents/utils';
+import {
+  createIncidentNote,
+  deleteIncidentNote,
+  fetchIncidentActivities,
+  updateIncidentNote,
+} from 'app/actionCreators/incident';
+import {t} from 'app/locale';
+import {uniqueId} from 'app/utils/guid';
+import ConfigStore from 'app/stores/configStore';
+import withApi from 'app/utils/withApi';
+
+import Activity from './activity';
+
+function makeDefaultErrorJson() {
+  return {detail: t('Unknown error. Please try again.')};
+}
+
+/**
+ * Activity component on Incident Details view
+ * Allows user to leave a comment on an incidentId as well as
+ * fetch and render existing activity items.
+ */
+class ActivityContainer extends React.Component {
+  static propTypes = {
+    api: PropTypes.object.isRequired,
+  };
+
+  state = {
+    loading: true,
+    error: false,
+    noteInputId: uniqueId(),
+    noteInputText: '',
+    createBusy: false,
+    createError: false,
+    activities: null,
+  };
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  async fetchData() {
+    const {api, params} = this.props;
+    const {incidentId, orgId} = params;
+
+    try {
+      const activities = await fetchIncidentActivities(api, orgId, incidentId);
+      this.setState({activities, loading: false});
+    } catch (err) {
+      this.setState({loading: false, error: !!err});
+    }
+  }
+
+  handleCreateNote = async note => {
+    const {api, params} = this.props;
+    const {incidentId, orgId} = params;
+
+    const newActivity = {
+      comment: note.text,
+      type: INCIDENT_ACTIVITY_TYPE.COMMENT,
+      dateCreated: new Date(),
+      user: ConfigStore.get('user'),
+      id: uniqueId(),
+      incidentIdentifier: incidentId,
+    };
+
+    this.setState(state => ({
+      createBusy: true,
```

Comment:
@markstory I've updated re: this comment https://github.com/getsentry/sentry/commit/575a2f05eba205baa769b170fe19d7faf4f356b2#r33669663